components/core/src/fs.rs

```diff
@@ -41,10 +36,24 @@ pub const CACHE_SRC_PATH: &'static str = "hab/cache/src";
 pub const CACHE_SSL_PATH: &'static str = "hab/cache/ssl";
 /// The root path containing all locally installed packages
 pub const PKG_PATH: &'static str = "hab/pkgs";
+/// The environment variable pointing to the filesystem root. This exists for internal
+/// Habitat team usage and is not intended to be used by Habitat consumers.
+/// Using this variable could lead to broken supervisor services and it should
+/// be used with extreme caution.
+pub const FS_ROOT_ENVVAR: &'static str = "FS_ROOT";
 /// The root path containing all runtime service directories and files
 const SVC_PATH: &'static str = "hab/svc";
 
 lazy_static! {
+    /// The default filesystem root path
+    pub static ref FS_ROOT_PATH: String = {
+        if cfg!(target_os = "windows") && henv::var(FS_ROOT_ENVVAR).is_ok() {
```

Comment:
I'm wondering what this would look like as 2 `pub static ref` statements with `#[cfg(…)]` attributes. The reason here is that I'm worried about the platform-dependant code getting tucked too far in and hidden. I realize that the 2 statements would look similar but am hopeful that it'll be super clear to any future development.